### PR TITLE
Closes #359: Truncate x-axis by default.

### LINKED ISSUE #359: Graphs are squashed with new redesign
The new redesign of Redash is very nice!  But some graphs are getting squashed and are difficult to read now, example:

https://sql.telemetry.mozilla.org/dashboard/activity-stream-system-addon-experiment-57-release

screenshot 2018-04-03 15 30 11

I think it's because the X axis labels aren't being truncated so the graph is being smushed.

### PR DESCRIPTION
**Pull request by emtwo**

### CODE REVIEW

File: client/app/visualizations/chart/index.js [hunk 1]
```diff
@@ -251,9 +251,18 @@ function ChartEditor(ColorPalette, clientConfig) {
         scope.options.legend = { enabled: true };
       }
 
-      if (!has(scope.options, 'xAxisLabelLength')) {
-        scope.options.xAxisLabelLength = 300;
-      }
+      scope.$watch('options.globalSeriesType', (newType, oldType) => {
+        const defaultXAxisLength = 10;
```
[the remark this answers is not in the document]
Comment (emtwo) [reply client/app/visualizations/chart/index.js]: This default is used when someone is making a visualization for the first time whereas `DEFAULT_XAXIS_LABEL_LENGTH` is the fallback default when someone clears out the value of the label `x-axis-label-length` (found in `chart-editor.html`) 

I think we can perhaps set `DEFAULT_XAXIS_LABEL_LENGTH` to 10 as well to be less confusing

[the remark this answers is not in the document]
Comment (washort) [reply client/app/visualizations/chart/index.js]: OK sounds good. On further reflection I wonder if pie charts and line charts should have different defaults.

[the remark this answers is not in the document]
Comment (emtwo) [reply client/app/visualizations/chart/index.js]: Yes, I would say there is probably a better/clearer way to set separate defaults for pie vs line charts. Though I realized that would be a bigger PR and need slightly more time (and we wanted to get this ready soon). For now this workaround does the trick to give pie charts a different default and I'm happy to file a follow-up.

[the remark this answers is not in the document]
Comment (washort) [reply client/app/visualizations/chart/index.js]: Great. r+

**Pull request merged**